### ISSUE TITLE
Combat animations happening too soon.

### ISSUE THREAD

**Issue by Flydx1234** [Tuxemon/Tuxemon#769]
These two issues are both related to combat animations/timing and may be related.

*[ ] Combat animations happen after the hp bar changes due to damage.
*[ ] After switching in a tuxemon, combat animations do not happen and damage occurs immediately to the tuxemon swapped in.

**Comment by bitcraft** [comment]
#1 is a known issue.  The issue with swapped monsters is interesting though.  What needs to be fixed, specifically?  The combat animations are lost?  There is a step roughly called "rewrite action queue", and its possible that that step needs to change the target of animations or something like that.

**Comment by Flydx1234** [comment]
Yeah when switching tuxemon, it looks like there's a missed frame or something and then your monster comes in damaged.

**Comment by Qiangong2** [comment]
The second one is most likely related to the first one. The HP isn't waiting until after the animation finishes to change
